pkg/datagatherer/aks/aks.go

```diff
@@ -11,19 +10,31 @@ import (
 	aks "github.com/Azure/aks-engine/pkg/api/agentPoolOnlyApi/v20180331"
 )
 
-// AKSDataGatherer is a DataGatherer for AKS.
-type AKSDataGatherer struct {
-	ctx           context.Context
-	resourceGroup string
-	clusterName   string
-	credentials   *AzureCredentials
+// Config is the configuration for an AKS DataGatherer.
+type Config struct {
+	// ClusterID is the ID of the cluster in AKS.
+	ClusterID string `mapstructure:"cluster-id"`
```

Comment:
I think in AKS they appear to use clustername rather than id https://docs.microsoft.com/en-us/rest/api/aks/managedclusters/get

ID seems to be the 'full qualified' id, e.g. `/subscriptions/subid1/resourcegroups/rg1/providers/Microsoft.ContainerService/managedClusters/clustername1`

Reply:
good point